sdk/eventhubs/azure-messaging-eventhubs/src/test/java/com/azure/messaging/eventhubs/EventHubClientBuilderTest.java

```diff
@@ -153,5 +159,47 @@ private static URI getURI(String endpointFormat, String namespace, String domain
         }
     }
 
+    @Test
+    public void shareConnectionTest() {
+        final EventHubClientBuilder builder = new EventHubClientBuilder();
+        builder.shareConnection();
+        Assertions.assertNotNull(builder);
+    }
+
+    @Test
+    public void transportTypeTest() {
+        final EventHubClientBuilder builder = new EventHubClientBuilder()
+            .transportType(AmqpTransportType.AMQP_WEB_SOCKETS);
+        Assertions.assertNotNull(builder);
+    }
+
+    @Test
+    public void retryTest() {
+        final EventHubClientBuilder builder = new EventHubClientBuilder()
+            .retry(retryOptions);
+        Assertions.assertNotNull(builder);
```

Comment:
Apart from checking for non-null return values of these builder methods, it's important to also verify that the client is created with the provided configuration. For e.g. if a retry option is set in the builder, the client created from this builder should be using the provided retry option instead of using the default option. The test should verify that.